x/shield/keeper/purchase.go

```diff
@@ -137,14 +147,25 @@ func (k Keeper) purchaseShield(ctx sdk.Context, poolID uint64, shield sdk.Coins,
 }
 
 // PurchaseShield purchases shield of a pool with standard fee rate.
-func (k Keeper) PurchaseShield(ctx sdk.Context, poolID uint64, shield sdk.Coins, description string, purchaser sdk.AccAddress) (types.Purchase, error) {
+func (k Keeper) PurchaseShield(ctx sdk.Context, poolID uint64, shield sdk.Coins, description string, purchaser sdk.AccAddress, staking bool) (types.Purchase, error) {
 	poolParams := k.GetPoolParams(ctx)
 	if poolParams.MinShieldPurchase.IsAnyGT(shield) {
 		return types.Purchase{}, types.ErrPurchaseTooSmall
 	}
 	bondDenom := k.BondDenom(ctx)
-	serviceFees := sdk.NewCoins(sdk.NewCoin(bondDenom, shield.AmountOf(bondDenom).ToDec().Mul(k.GetPoolParams(ctx).ShieldFeesRate).TruncateInt()))
-	return k.purchaseShield(ctx, poolID, shield, description, purchaser, serviceFees)
+	serviceFees := sdk.NewCoins()
+	stakingCoins := sdk.NewCoins()
+	if !staking {
+		serviceFees = sdk.NewCoins(sdk.NewCoin(bondDenom, shield.AmountOf(bondDenom).ToDec().Mul(k.GetPoolParams(ctx).ShieldFeesRate).TruncateInt()))
+	} else {
+		// stake to the staking purchase pool
+		stakingAmt := k.GetStakeForShieldRate(ctx).MulInt(shield.AmountOf(bondDenom)).TruncateInt()
+		stakingCoins = sdk.NewCoins(sdk.NewCoin(bondDenom, stakingAmt))
+		if err := k.supplyKeeper.SendCoinsFromAccountToModule(ctx, purchaser, types.ModuleName, stakingCoins); err != nil {
```

Comment:
Isn't this a duplicate of what's in `AddStaking()`?

Reply:
Yeah it was. Good catch